tests/java/iceberg-rest/src/main/java/org/apache/iceberg/rest/RESTCatalogServer.java

```diff
@@ -0,0 +1,136 @@
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RESTCatalogServer {
+  private static final Logger LOG
+      = LoggerFactory.getLogger(RESTCatalogServer.class);
+  private static final String CATALOG_ENV_PREFIX = "CATALOG_";
+  // absolute file path from which hadoop configuration is loaded, if set.
+  private static final String HADOOP_CONF_FILE_PATH = "HADOOP_CONF_FILE_PATH";
+
+  private RESTCatalogServer() {}
+
+  record CatalogContext(Catalog catalog, Map<String, String> configuration) {}
+
+  private static CatalogContext backendCatalog() throws IOException {
+    // Translate environment variable to catalog properties
+    Map<String, String> catalogProperties
+        = System.getenv()
+              .entrySet()
+              .stream()
+              .filter(e -> e.getKey().startsWith(CATALOG_ENV_PREFIX))
+              .collect(Collectors.toMap(
+                  e
+                  -> e.getKey()
+                         .replaceFirst(CATALOG_ENV_PREFIX, "")
+                         .replaceAll("__", "-")
+                         .replaceAll("_", ".")
+                         .toLowerCase(Locale.ROOT),
+                  Map.Entry::getValue, (m1, m2) -> {
+                    throw new IllegalArgumentException("Duplicate key: " + m1);
+                  }, HashMap::new));
+
+    // Fallback to a JDBCCatalog impl if one is not set
+    catalogProperties.putIfAbsent(
+        CatalogProperties.CATALOG_IMPL, "org.apache.iceberg.jdbc.JdbcCatalog");
+    catalogProperties.putIfAbsent(
+        CatalogProperties.URI,
+        "jdbc:sqlite:file:/tmp/iceberg_rest_mode=memory");
+    catalogProperties.putIfAbsent("jdbc.schema-version", "V1");
+
+    // Configure a default location if one is not specified
+    String warehouseLocation
+        = catalogProperties.get(CatalogProperties.WAREHOUSE_LOCATION);
+
+    if (warehouseLocation == null) {
+      File tmp = java.nio.file.Files.createTempDirectory("iceberg_warehouse")
+                     .toFile();
+      tmp.deleteOnExit();
+      warehouseLocation
+          = tmp.toPath().resolve("iceberg_data").toFile().getAbsolutePath();
+      catalogProperties.put(
+          CatalogProperties.WAREHOUSE_LOCATION, warehouseLocation);
+
+      LOG.info(
+          "No warehouse location set.  Defaulting to temp location: {}",
+          warehouseLocation);
+    }
+
+    LOG.info("Creating catalog with properties: {}", catalogProperties);
+
+    Configuration conf = new Configuration();
+    String hadoopConfPath = System.getenv(HADOOP_CONF_FILE_PATH);
+    if (hadoopConfPath != null) {
+      File confFile = new File(hadoopConfPath);
+      if (confFile.exists() && !confFile.isDirectory()) {
+        LOG.info("Loading hadoop configuration from path: " + confFile.toURI());
+        conf.addResource(new Path(confFile.toURI()));
+      }
```

Comment:
changed by me: to load custom hadoop core-site.xml at runtime (for some reason it doesn't load one from the classpath) 🤷